tensorflow/lite/kernels/comparisons.cc

```diff
@@ -37,9 +37,8 @@ TfLiteStatus ComparisonPrepare(TfLiteContext* context, TfLiteNode* node) {
   const TfLiteTensor* input2 = GetInput(context, node, kInputTensor2);
   TfLiteTensor* output = GetOutput(context, node, kOutputTensor);
 
-  // Don't support string and bool.
-  TF_LITE_ENSURE(context,
-                 input1->type != kTfLiteString || input1->type != kTfLiteBool);
+  // Don't support string.
+  TF_LITE_ENSURE(context, input1->type != kTfLiteString);
```

Comment:
the problem is this compare prepare is shared among (== , !=, >=, <=, >, <), and it seems bool can only support == & !=

Reply:
> the problem is this compare prepare is shared among (== , !=, >=, <=, >, <), and it seems bool can only support == & !=

@renjie-liu : Yes, you are right, i also observed that. But i think as the wrong datatype can also be captured in Eval() stage and it is not a costly operation, so i felt, lets keep the Prepare() more generic and excluded any specific case check in Prepare().

Welcome for any suggestions you have, Thanks!